src/messaging/messaging.ts

```diff
@@ -332,6 +339,40 @@ function validateAps(aps: Aps) {
   }
 }
 
+function validateApsSound(sound: string | CriticalSound) {
+  if (typeof sound === 'undefined' || validator.isString(sound)) {
+    return;
+  } else if (!validator.isNonNullObject(sound)) {
+    throw new FirebaseMessagingError(
+      MessagingClientErrorCode.INVALID_PAYLOAD,
+      'apns.payload.aps.sound must be a string or a non-null object');
+  }
+
+  const volume = (sound as CriticalSound).volume;
+  if (typeof volume !== 'undefined') {
+    if (!validator.isNumber(volume)) {
+      throw new FirebaseMessagingError(
+        MessagingClientErrorCode.INVALID_PAYLOAD,
+        'apns.payload.aps.sound.volume must be a number');
+    }
+    if (volume < 0 || volume > 1) {
+      throw new FirebaseMessagingError(
+        MessagingClientErrorCode.INVALID_PAYLOAD,
+        'apns.payload.aps.sound.volume must be in the interval [0, 1]');
+    }
+  }
+  const soundObject = sound as {[key: string]: any};
+  const key = 'critical';
+  const critical = soundObject[key];
+  if (typeof critical !== 'undefined' && critical !== 1) {
```

Comment:
We should not need to check `critical !== 1` since it's defined as a boolean.

Reply:
JS developers often pass 1 for this kind of values, since APNs documentation says so. Our policy so far has been to silently allow it.

Reply:
hmm, that's interesting. Any reason we deviate from the APNs definition? Can we just define critical as number?

Reply:
Well, the only acceptable values for this field are 0 and 1. So it's really a bool that APNs exposes as an int for some reason. We have always exposed them as bools in our public API, but been flexible to accept ints as well. See the `Aps.mutableContent` property for a similar field.